src/tests/TaskList.test.tsx

```diff
@@ -0,0 +1,76 @@
+// src/tests/TaskList.test.tsx
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import TaskList from '../components/TaskList';
+
+describe('Task Management App', () => {
+    test('allows user to add a new task with category and priority', () => {
+        render(<TaskList />);
+
+        const taskInput = screen.getByPlaceholderText(/Task description/i);
+        const categoryInput = screen.getByPlaceholderText(/Task Category/i);
+        const prioritySelect = screen.getByDisplayValue('Low Priority');
+        const addButton = screen.getByRole('button', { name: /Add Task/i });
+
+        fireEvent.change(taskInput, { target: { value: 'New Task' } });
```

Comment:
you can make use of faker to generate data for tests instead of using static test data

Reply:
fixed